File: bindings/nodejs/lib/types/utils/decayed-mana.ts
// Copyright 2024 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { u64 } from './type-aliases';

/** Decayed stored and potential Mana of an output. */
export interface DecayedMana {
    /** Decayed stored mana. */
    stored: u64;
    /** Decayed potential mana. */
    potential: u64;
}

File: bindings/python/iota_sdk/types/decayed_mana.py
# Copyright 2024 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

from dataclasses import dataclass
from iota_sdk.types.common import json


@json
@dataclass
class DecayedMana:
    """Decayed stored and potential Mana of an output.

        Attributes:
        stored (int):
            Decayed stored mana.
        potential (int):
            Decayed potential mana.
    """

    stored: int
    potential: int
